src/app/common/utilities/helpers.ts

```
@@ -58,4 +60,45 @@ function createInitialArray(element, count) {
   return resultArray;
 }
 
-export { cloneDeep, enumToArray, enumToObject, enumToObjectArray, isUnixTimePastNow, selectCompare, createInitialArray };
\ No newline at end of file
+function sortArrayByColumnMetaData(sortArray, columnMetaData: TableCoulmnMetaData) {
+  if (!columnMetaData.keys || !columnMetaData.hasOwnProperty('sortAsc')) {
+    return;
```

Do you really want to return undefined here? This might lead to tricky debugging later on. If you're returning undefined here just because the config is wrong, I'd rather throw an error so that you'd see what's wrong right away.

---

Not every column is a category for sort, here when some fields are missing in meta data code prevents sorting. So throwing error might be overreacting

---

Valid point. You have changed the implementation to return a sorted array, so you have to return the initial array here instead of returning `undefined`.

---

good point